pkg/reconciliation/reconcile_racks.go

```diff
@@ -1806,97 +1809,99 @@ func (rc *ReconciliationContext) startCassandra(endpointData httphelper.CassMeta
 	return rc.labelServerPodStarting(pod)
 }
 
-// returns the name of one rack without any ready node
-func (rc *ReconciliationContext) startOneNodePerRack(endpointData httphelper.CassMetadataEndpoints, readySeeds int) (string, error) {
-
+// startOneNodePerRack starts the first pod in each rack in a deterministic order, dictated by the
+// order in which the racks were declared in the datacenter spec. It returns true if a Cassandra
+// node is not ready yet, and an error if that node also failed to start.
+func (rc *ReconciliationContext) startOneNodePerRack(endpointData httphelper.CassMetadataEndpoints, readySeeds int) (notReady bool, err error) {
 	rc.ReqLogger.Info("reconcile_racks::startOneNodePerRack")
 
-	rackReadyCount := map[string]int{}
-	for _, rackInfo := range rc.desiredRackInformation {
-		rackReadyCount[rackInfo.RackName] = 0
-	}
+	labelSeedBeforeStart := readySeeds == 0 && !rc.hasAdditionalSeeds()
 
-	for _, pod := range rc.dcPods {
-		rackName := pod.Labels[api.RackLabel]
-		if isServerReady(pod) {
-			rackReadyCount[rackName]++
-		}
-	}
+	for _, statefulSet := range rc.statefulSets {
 
-	// if the DC has no ready seeds, label a pod as a seed before we start Cassandra on it
-	// and also consider additional seeds
+		pod := rc.getDCPodByName(statefulSet.Name + "-0")
+		notReady, err = rc.startNode(pod, labelSeedBeforeStart, endpointData)
+		if notReady {
+			break
+		}
 
-	externalSeedPoints := 0
-	if existingEndpoints, err := rc.GetAdditionalSeedEndpoint(); err == nil {
-		externalSeedPoints = len(existingEndpoints.Subsets[0].Addresses)
 	}
+	return
```

Comment:
The named returns serve no purpose here, they only make reading more difficult. They're not used in cass-operator source code anywhere else.

Reply:
I obviously think the opposite, otherwise I wouldn't have used named return parameters in the first place :-)

But if you don't like them, let's remove them.